src/axom/slam/Map.hpp

```diff
@@ -409,58 +550,148 @@ class Map : public StrPol,
    *          the currently pointed to element (where 0 <= j < numComp()).\n
    *          For example: `iter[off]` is the same as `(iter+off)(0)`
    */
-  class MapIterator : public IteratorBase<MapIterator, SetPosition>
+  template <bool Const>
+  class MapRangeIterator
+    : public IteratorBase<MapRangeIterator<Const>, SetPosition>
   {
   public:
+    using IterBase = IteratorBase<MapRangeIterator, SetPosition>;
+    using MapConstPtr = std::conditional_t<Const, const Map*, Map*>;
+
+    using DataRefType = std::conditional_t<Const, ConstValueType, ValueType>;
+    using DataType = std::remove_reference_t<DataRefType>;
+
+    using PositionType = SetPosition;
+    using StrideIndexType = typename StridePolicyType::IndexType;
+    constexpr static int Dims = StridePolicyType::NumDims;
+
+    // Type traits to satisfy LegacyRandomAccessIterator concept
     using iterator_category = std::random_access_iterator_tag;
-    using value_type = DataType;
+    using value_type = axom::ArrayView<DataType, Dims>;
+    using reference = const value_type&;
+    using pointer = const value_type*;
     using difference_type = SetPosition;
 
-    using IterBase = IteratorBase<MapIterator, SetPosition>;
-    using iter = MapIterator;
-    using PositionType = SetPosition;
-    using IterBase::m_pos;
+  private:
+    static StackArray<IndexType, Dims + 1> fetchDims(StrideIndexType stride)
```

Comment:
This overload only matches when `StridePolicy` is one of the 1D policies, `RuntimeStride` or `CompileTimeStride`. In that case, the value returned from `StridePolicyType::shape()` is a scalar int type, instead of a `StackArray` as with the `MultiDimStride` policy.

I'll rename the `stride` parameter to `shape` to correspond with its usage.